adam/learner/subset.py

```diff
@@ -44,10 +49,23 @@ def _learning_step(
             previous_pattern_hypothesis = self._surface_template_to_hypothesis[
                 surface_template
             ]
+            raw_new_pattern_hypothesis = self._hypothesis_from_perception(
+                preprocessed_input
+            )
 
             updated_hypothesis = previous_pattern_hypothesis.intersection(
-                self._hypothesis_from_perception(preprocessed_input),
+                raw_new_pattern_hypothesis,
                 ontology=self._ontology,
+                # Q: Is there a better way to compute this? Is it guaranteed that the slots are
+                # in order so that we can do zip(prev, new)?
```

Comment:
I don't think it's strictly guaranteed (as in by a `attrs_post_init` or something similar) that the slots are in order however they should be because we only pull the slots in order.

Reply:
Thanks. In that case, I think it makes sense to keep it like this. I'll remove the comment.